src/ucp/wireup/wireup_cm.c

```diff
@@ -102,15 +158,45 @@ static void ucp_cm_priv_data_pack(ucp_wireup_sockaddr_data_t *sa_data,
     memcpy(sa_data + 1, addr, addr_size);
 }
 
+static void uct_wireup_cm_tmp_ep_cleanup(ucp_ep_h tmp_ep, ucs_queue_head_t *queue)
+{
+    ucp_lane_index_t lane_idx;
+    uct_ep_h uct_ep;
+
+    for (lane_idx = 0; lane_idx < ucp_ep_num_lanes(tmp_ep); ++lane_idx) {
+        if (lane_idx == ucp_ep_get_cm_lane(tmp_ep)) {
+            continue;
+        }
+
+        /* transfer the pending queues content from the previous tmp_ep to
+         * a temporary queue */
+        uct_ep_pending_purge(tmp_ep->uct_eps[lane_idx],
+                             ucp_wireup_pending_purge_cb, &queue);
+
+        if (ucp_ep_config(tmp_ep)->p2p_lanes & UCS_BIT(lane_idx)) {
+            uct_ep = ucp_wireup_extract_lane(tmp_ep, lane_idx);
+            /* destroy the transport ep */
+            uct_ep_destroy(uct_ep);
```

Comment:
can it be united with CM lane into common function?

Reply:
can you please clarify? unite these lines with what?

Reply:
unset_ep and this code do similar things